internal/provider/resource_mackerel_service_test.go

```diff
@@ -24,3 +36,145 @@ func TestMackerelServiceResourceSchema(t *testing.T) {
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { preCheck(t) },
+		ProtoV5ProviderFactories: protoV5ProviderFactories(),
+		CheckDestroy:             checkMackerelServiceDestroy(client),
+		Steps: []resource.TestStep{
+			// Create
+			{
+				Config: mackerelServiceResourceConfig_basic(name),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(resourceName, plancheck.ResourceActionCreate),
+						plancheck.ExpectUnknownValue(resourceName, tfjsonpath.New("id")),
+						plancheck.ExpectKnownValue(resourceName, tfjsonpath.New("name"), knownvalue.StringExact(name)),
+						plancheck.ExpectKnownValue(resourceName, tfjsonpath.New("memo"), knownvalue.Null()),
+					},
+				},
+				ConfigStateChecks: []statecheck.StateCheck{
+					expectMackerelService(client, resourceName, mackerel.Service{
+						Name: name,
+						Memo: "",
+					}),
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("id"), knownvalue.StringExact(name)),
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("name"), knownvalue.StringExact(name)),
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("memo"), knownvalue.Null()),
+				},
+			},
+			// Update
+			{
+				Config: mackerelServiceResourceConfig_withMemo(nameUpdated, memoUpdated),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(resourceName, plancheck.ResourceActionDestroyBeforeCreate),
+						plancheck.ExpectUnknownValue(resourceName, tfjsonpath.New("id")),
+						plancheck.ExpectKnownValue(resourceName, tfjsonpath.New("name"), knownvalue.StringExact(nameUpdated)),
+						plancheck.ExpectKnownValue(resourceName, tfjsonpath.New("memo"), knownvalue.StringExact(memoUpdated)),
+					},
+				},
+				ConfigStateChecks: []statecheck.StateCheck{
+					expectMackerelService(client, resourceName, mackerel.Service{
+						Name: nameUpdated,
+						Memo: memoUpdated,
+					}),
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("id"), knownvalue.StringExact(nameUpdated)),
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("name"), knownvalue.StringExact(nameUpdated)),
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("memo"), knownvalue.StringExact(memoUpdated)),
+				},
+			},
+			// Import
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func mackerelServiceResourceConfig_basic(name string) string {
+	return fmt.Sprintf(`
+resource "mackerel_service" "foo" {
+  name = "%s"
+}`, name)
+}
+
+func mackerelServiceResourceConfig_withMemo(name string, memo string) string {
+	return fmt.Sprintf(`
+resource "mackerel_service" "foo" {
+  name = "%s"
+  memo = %q
+}`, name, memo)
+}
+
+func checkMackerelServiceDestroy(client *mackerel.Client) resource.TestCheckFunc {
```

Comment:
[State checks](https://developer.hashicorp.com/terraform/plugin/testing/acceptance-tests/state-checks) are new feature and currently have no equivalence for `CheckDestroy`, so `resource.TestCheckFunc` is used.